### PR TITLE
[FIX] account_move_reconcile_forbid_cancel: make it compatible with tests of other modules

### PR DESCRIPTION
**Pull request by ernestotejeda**
TT37518

- Make it compatible with tests of other modules
- Fix payable account type detection and add tests for vendor invoice

### CODE REVIEW

**Comment by ernestotejeda** [comment]
Ping @carlosdauden @pedrobaeza

File: account_move_reconcile_forbid_cancel/models/account_move.py [hunk 1]
```diff
@@ -10,17 +10,25 @@ class AccountMove(models.Model):
 
     def _get_receivable_payable_lines(self):
         return self.line_ids.filtered(
-            lambda l: l.account_internal_type in ["receivable", "acc_type_pay"],
+            lambda l: l.account_internal_type in ["receivable", "payable"],
```
Comment (pedrobaeza) [inline account_move_reconcile_forbid_cancel/models/account_move.py:13]: Why this change?

Comment (ernestotejeda) [reply account_move_reconcile_forbid_cancel/models/account_move.py:13]: This is another fix. I have extracted it to another commit.

**Comment by ernestotejeda** [comment]
I splitted the commit into 2 commit to have each functionality in a different commit.
ping @pedrobaeza @carlosdauden

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Fast-tracking this to fix the bug and the integration tests:

/ocabot merge patch
